src/ReadComicsTo/Parser.ts

```diff
@@ -0,0 +1,170 @@
+    released = this.decodeHTMLEntity($('p:has(span:contains(Publication date:))', infoElement).first().text()).replace('Publication date:', '').trim()
+    
+    let statusViewsParagraph = $('p:has(span:contains(Status:))', infoElement).first().text().toLowerCase()
+    status = statusViewsParagraph.includes('ongoing') ? MangaStatus.ONGOING : MangaStatus.COMPLETED
+    views = Number(statusViewsParagraph.replace('\n', '').split('\n')[1]?.trim()?.replace(/\D/g, ''))
+
+    let genres = $('p:has(span:contains(Genres:)) > a', infoElement).toArray()
+    for (let obj of genres) {
+        let id = $(obj).attr('href')
+        let label = this.decodeHTMLEntity($(obj).text().trim())
+        if (typeof id === 'undefined' || typeof label === 'undefined') continue
+        tagArray0 = [...tagArray0, createTag({id: id, label: label})]
+    }
+    let tagSections: TagSection[] = [createTagSection({ id: '0', label: 'genres', tags: tagArray0 })]
+      return createManga({
+        id: mangaId,
+        rating: rating,
+        titles: titles,
+        image: image ?? '',
+        status: status,
+        author: this.decodeHTMLEntity(author ?? ''),
+        artist: artist,
+        views: views,
+        tags: tagSections,
+        desc: this.decodeHTMLEntity(summary ?? ''),
+        lastUpdate: released
+      })
+    }
+   
+
+    parseChapterList($: CheerioSelector, mangaId: string) : Chapter[] { 
+    
+    let chapters: Chapter[] = []
+
+    let chapArray = $('tr', $('.listing').first()).toArray().reverse()
+    for(let i = 0; i < chapArray.length; i++) {
+           let obj = chapArray[i]
+           let chapterId = $('a', obj)?.first().attr('href')?.replace(`/Comic/${mangaId}/`, '')
+           let chapter = $('td', obj).first()
+           let chapNum = i + 1
+           let chapName = chapter.text().trim()
+           let time = $('td', $(obj)).last().text().trim()
+           if (typeof chapterId === 'undefined') continue
+           chapters.push(createChapter({
+               id: chapterId,
+               mangaId: mangaId,
+               chapNum: Number(chapNum),
+               langCode: LanguageCode.ENGLISH,
+               name: this.decodeHTMLEntity(chapName),
+               time: new Date(time)
+           }))
+       }
+       return chapters
+}
+
+
+
+
+
+    parseChapterDetails(data: string) : string[] {
+        
+      let pages = [...data.matchAll(/lstImages\.push\("(http.*)"\)/g)]
+      return pages.map(match => match[1])
+    }
+
+
+    parseSearchResults($: CheerioSelector, cheerio: any): MangaTile[] { 
+        let mangaTiles: MangaTile[] = []
+        let collectedIds: string[] = []
+        for(let obj of $('tr', $('.listing')).toArray()) {
+            
+            let titleText = this.decodeHTMLEntity($('a',$(obj)).text().replace('\n','').trim())
+            let id = $('a',$(obj)).attr('href')?.replace('/Comic/', '')
+            if(!titleText) { 
```

Comment:
There's some awkward spacing below (optional fix)
I suggest also early-exiting if you can't find an ID though.

```suggestion
            if(!titleText || !id) { 
```